src/main/kotlin/org/pkl/intellij/util/Feature.kt

```diff
@@ -17,17 +17,28 @@ package org.pkl.intellij.util
 
 import com.intellij.psi.PsiElement
 import org.pkl.intellij.PklVersion
-import org.pkl.intellij.psi.PklModule
+import org.pkl.intellij.psi.*
 
-sealed class Feature(
-  val featureName: String,
-  val requiredVersion: PklVersion,
+sealed interface Feature {
+  val featureName: String
+  val requiredVersion: PklVersion
   val predicate: (PsiElement) -> Boolean
-) {
+  val message: String?
+    get() = null
 
   fun isSupported(module: PklModule): Boolean = module.effectivePklVersion >= requiredVersion
```

Comment:
```suggestion
  fun isSupported(module: PklModule): Boolean = module.effectivePklVersion >= requiredVersion

  fun featureMessage(): String = message ?: "Unsupported feature: $featureName"
```

^^ Seems reusable enough